src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportProcessingJob.cs

```diff
@@ -95,17 +97,27 @@ record = exportJobRecord;
                     // This will allow for saving intermediate states of the job.
 
                     record.Status = OperationStatus.Completed;
-                    var existingJobs = await _queueClient.GetJobByGroupIdAsync(QueueType.Export, jobInfo.GroupId, false, cancellationToken);
+                    var existingJobs = await _queueClient.GetJobByGroupIdAsync(QueueType.Export, jobInfo.GroupId, false, innerCancellationToken);
 
-                    // Checks that a followup job has not already been made.
-                    if (!existingJobs.Any((job) => job.Id > jobInfo.Id))
+                    // Only queue new jobs if group has no canceled jobs. This ensures canceled jobs won't continue to queue new jobs (Cosmos cancel is not 100% reliable).
+                    if (!existingJobs.Any(job => job.Status == JobStatus.Cancelled || job.CancelRequested))
                     {
                         var definition = jobInfo.DeserializeDefinition<ExportJobRecord>();
-                        definition.Progress = exportJobRecord.Progress;
-                        await _queueClient.EnqueueAsync(QueueType.Export, innerCancellationToken, jobInfo.GroupId, definitions: definition);
+
+                        // Checks that a follow up job has not already been made. Extra checks are needed for parallel jobs by parallelization factors.
+                        bool newerJobsExist = existingJobs.Where(existingJob => existingJob.Definition is not null).Any(existingJob =>
+                        {
+                            var existingDefinition = existingJob.DeserializeDefinition<ExportJobRecord>();
+                            return existingJob.Id > jobInfo.Id && existingDefinition.ResourceType == definition.ResourceType && existingDefinition.FeedRange == definition.FeedRange;
+                        });
+
+                        if (!newerJobsExist)
+                        {
+                            definition.Progress = exportJobRecord.Progress;
+                            await _queueClient.EnqueueAsync(QueueType.Export, innerCancellationToken, jobInfo.GroupId, definitions: definition);
+                        }
                     }
 
-                    // TODO: When legacy export support is removed from Cosmos remove this exception and refactor ExportJobTask to expect only one page of results will be processed.
```

Comment:
I would still like to clean this up at some point, but I guess Cosmos being removed will happen sooner than that.

Reply:
I agree - though our current data schema with Cosmos current GA features means the exported rows would need to be traversed twice. The current implementation is what the Cosmos team recommended but open to change it in the future (or now if it's important).

If our date/time format was true ISO 8601, we could use a swath of date time functions (like `DateTimeBin`) that would approach SQL windowing.